src/lang/lang.ml

```diff
@@ -161,24 +161,48 @@ let compare_values a b =
     | Ground a, Ground b -> Ground.compare a b
     | Tuple l, Tuple m ->
         List.fold_left2
-          (fun cmp a b -> if cmp <> 0 then cmp else aux (a.value, b.value))
+          (fun cmp a b -> if cmp <> 0 then cmp else compare_values (a, b))
           0 l m
     | List l1, List l2 ->
         let rec cmp = function
           | [], [] -> 0
           | [], _ -> -1
           | _, [] -> 1
           | h1 :: l1, h2 :: l2 ->
-              let c = aux (h1.value, h2.value) in
+              let c = compare_values (h1, h2) in
               if c = 0 then cmp (l1, l2) else c
         in
         cmp (l1, l2)
     | Null, Null -> 0
     | Null, _ -> -1
     | _, Null -> 1
     | _ -> assert false
+  and compare_values (a, b) =
```

Comment:
This would better be moved to `Value.compare`.

Reply:
This is really annoying to do in the `2.0.0` branch as `Value` is nested into `Term` and the namings clash. I'll do it when backporting to `main`.